### PR TITLE
[flutter_tools] move gradle download failure handling into tool

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

This crash affects anyone using flutter, so move it into the tool where it can be tested and documented. Ensure it works on windows devicelab too.

### CODE REVIEW

File: dev/devicelab/lib/framework/runner.dart [hunk 1]
```diff
@@ -161,10 +161,5 @@ Future<void> cleanupSystem() async {
     } else {
       print('Could not determine JAVA_HOME; not shutting down Gradle.');
     }
-    // Removes the .gradle directory because sometimes gradle fails in downloading
-    // a new version and leaves a corrupted zip archive, which could cause the
-    // next devicelab task to fail.
-    // https://github.com/flutter/flutter/issues/65277
-    rmTree(dir('${Platform.environment['HOME']}/.gradle'));
```
Comment (jonahwilliams) [inline dev/devicelab/lib/framework/runner.dart:168]: this also doesn't work on windows

File: packages/flutter_tools/lib/src/android/gradle_errors.dart [hunk 2]
```diff
@@ -120,6 +126,15 @@ final GradleHandledError networkErrorHandler = GradleHandledError(
       '$warningMark Gradle threw an error while downloading artifacts from the network. '
       'Retrying to download...'
     );
+    try {
+      final String homeDir = globals.platform.environment['HOME'];
```
Comment (dnfield) [inline packages/flutter_tools/lib/src/android/gradle_errors.dart:130]: Wouldn't this be in %LOCALAPPDATA% or whatever on Windows? Can we try to check there?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/android/gradle_errors.dart:130]: Its in HOME for me

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/android/gradle_errors.dart:130]: Ahh, its even documented: https://docs.gradle.org/current/userguide/directory_layout.html#dir:gradle_user_home

**Review by dnfield — APPROVED** [review]
LGTM

**Comment by AlexV525** [comment]
Deleting the whole `.gradle` actually makes Flutter run much slower in nowadays Flutter development. Flutter is not the only one that uses `.gradle`. It might be too aggressive to handle like this. Would we be able to clean the specific failed file or try to narrow down the cleaning scope?

**Comment by AlexV525** [comment]
Or make a flag for devices in the devicelab that can delete `.gradle`?

**Pull request merged**